### ISSUE TITLE
Setting the `host` header on the request is not reflected on the proxied request.

### ISSUE THREAD

**Issue by fbuetler** [eclipse-vertx/vertx-http-proxy#85]
### Version

`4.5.8`

### Context

When using the `http-proxy` and setting the `Host` header, it depends on whether to set the header as `Host` or `host`. With `Host`, the value is correctly reflected on the server, the proxy is sending the request to. But if it is `host`, the header value is not present on the request at the server.

By RFC 9110, header (or field) names are case-insensitive:

> Field names are case-insensitive

(Source: https://www.rfc-editor.org/rfc/rfc9110.html#fields)

### Do you have a reproducer?

A reproducer is attached as a zip. It is generated with the `vertx-starter` uses `vertx-web` and `vertx-http-proxy`.

[reproducer.zip](https://github.com/user-attachments/files/15954364/reproducer.zip)

### Steps to reproduce

1. `mvn test`
2. observe 1 out of 2 test failures.

**Comment by fbuetler** [comment]
Upon further inspection, it is probably related to this line here https://github.com/eclipse-vertx/vertx-http-proxy/blob/4.5.8/src/main/java/io/vertx/httpproxy/impl/ProxiedRequest.java#L172

Why is the proxy not allowing to set the `host` header? Should another instance later set the correct header value?

**Comment by tsegismont** [comment]
Hi @fbuetler , thanks for reporting this issue.

The case-sensitivity problem has been solved in https://github.com/eclipse-vertx/vertx-http-proxy/pull/82

The host header shouldn't be preserved by default, if you want the value to be conveyed to the backend, invoke the `setAuthority` method on the `ProxiedRequest` in an interceptor. In this case, the value will be sent as `x-forwarded-host` header.

**Comment by fbuetler** [comment]
I think the default for the outgoing request should not be taken from the original request [1], but from the request to be sent out [2]. As the request, to be sent out, has the correct host set on its creation [3]

[1] https://github.com/eclipse-vertx/vertx-http-proxy/blob/4.5.8/src/main/java/io/vertx/httpproxy/impl/ProxiedRequest.java#L81
[2] https://github.com/eclipse-vertx/vertx-http-proxy/blob/4.5.8/src/main/java/io/vertx/httpproxy/impl/ProxiedRequest.java#L226
[3] https://github.com/eclipse-vertx/vert.x/blob/4.5.8/src/main/java/io/vertx/core/http/impl/HttpClientImpl.java#L380

**Comment by fbuetler** [comment]
@tsegismont can you please have another look at this and reopen the issue? I still think this is an issue.

**Comment by tsegismont** [comment]
> I agree that the host header should not be preserved. However, currently the host header is left untouched in the `vertx-http-proxy` and the **original host header, sent to the proxy, is forwarded to the target host**.

Yes, there is no released version that fixes this bug yet, but the issue is tracked by #77 and has been solved in the `4.x` branch by #79.

In Vert.x 4.5.9, the host header will no longer be preserved, regardless of the case.

**Comment by fbuetler** [comment]
As the #77 is now included in a released version i.e. 4.5.9, I revisited this issue and upgraded my reproducer to 4.5.14 and this still is an issue: The host header of the request arriving to a backend server is still set to the initial host i.e. the proxy host. The host header is still preserved by default.

[reproducer-updated.zip](https://github.com/user-attachments/files/19895408/reproducer-updated.zip)

**Comment by tsegismont** [comment]
Thanks for the detailed analysis

**Comment by tsegismont** [comment]
Fixed by #119
